src/cpl/share_esmf/PrigentRoughnessStreamType.F90

```diff
@@ -0,0 +1,335 @@
+    !
+    ! !DESCRIPTION:
+    ! Return true if the Prigent Roughness stream is being used
+    !
+    ! !USES:
+    !
+    ! !ARGUMENTS:
+    implicit none
+    class(prigentroughnessstream_type) :: this
+    !
+    character(len=*), parameter :: subname = 'PrigentRoughnessStream::UseStreams'
+    !
+    if ( .not. NMLRead )then
+       call endrun(msg=subname//' ERROR Namelist has NOT been read first, call Init before this')
+    end if
+    if ( trim(control%stream_fldFileName_prigentroughness) == '' )then
+       UseStreams = .false.  ! Prigent streams are off without a filename given
+    else
+       UseStreams = .true.
+    end if
+  end function UseStreams
+
+  !==============================================================================
+  logical function IsStreamInit(this)
+    !
+    ! !DESCRIPTION:
+    ! Return true if the streams have been initialized
+    !
+    ! !USES:
+    !
+    ! !ARGUMENTS:
+    implicit none
+    class(prigentroughnessstream_type) :: this
+    !
+    character(len=*), parameter :: subname = 'PrigentRoughnessStream::IsStreamInit'
+    !
+    if ( .not. NMLRead )then
+       call endrun(msg=subname//' ERROR Namelist has NOT been read first, call Init before this')
+    end if
+    if ( InitDone )then
+       IsStreamInit = .true.
+    else
+       IsStreamInit = .false.
+    end if
+  end function IsStreamInit
+
+  !==============================================================================
+  subroutine InitAllocate(this, bounds)
+    !
+    ! !DESCRIPTION:
+    ! Allocate module variables and data structures
+    !
+    ! !USES:
+    use shr_infnan_mod, only: nan => shr_infnan_nan, assignment(=)
+    !
+    ! !ARGUMENTS:
+    implicit none
+    class(prigentroughnessstream_type) :: this
+    type(bounds_type), intent(in) :: bounds
+    !
+    ! !LOCAL VARIABLES:
+    integer  :: begg, endg
+    !---------------------------------------------------------------------
+
+    begg = bounds%begg; endg = bounds%endg
+
+    if ( this%useStreams() )then
+       allocate(this%prigent_rghn(begg:endg))
+    else
+       allocate(this%prigent_rghn(0))
+    end if
+    this%prigent_rghn(:) = nan
+
+  end subroutine InitAllocate
+
+  !==============================================================================
+  subroutine ReadNML(this, bounds, NLFilename)
+   !
+   ! Read the namelist data stream information.
+   !
+   ! Uses:
+   use shr_nl_mod       , only : shr_nl_find_group_name
+   use shr_log_mod      , only : errMsg => shr_log_errMsg
+   use shr_mpi_mod      , only : shr_mpi_bcast
+   !
+   ! arguments
+   implicit none
+   class(streamcontrol_type)     :: this
+   type(bounds_type), intent(in) :: bounds
+   character(len=*),  intent(in) :: NLFilename   ! Namelist filename
+   !
+   ! local variables
+   integer            :: nu_nml    ! unit for namelist file
+   integer            :: nml_error ! namelist i/o error flag
+   logical            :: use_prigent_roughness = .true.
```

Comment:
Curious why do we need this logical variable? This has to be true always and can't be false in Leung. Is this mainly because Zender doesn't need the Prigent stream? I though Meier needs this too though.

Reply:
@dmleung good question. There are several reasons for this. To help make the namelist handling easier, to facilitate Paleo work, and to allow it's usage for things outside of Dust (like Meier for example).

First of all in the build-namelist call it's useful to have a logical variable that can be set to determine if the roughness stream needs to be set. With the logical variable the build-namelist can set the filenames for the user by default. It also allows some error checking to be done to make sure namelist variables are consistent.

The second reason though is that we'll need to allow turning roughness off for Paleo cases. Palo cases could also provide their own roughness stream, but I think this data will be too hard to guess for many Paleo situations so we'll just need a simple way to turn it off.

Yes, and the last reason is because this dataset could be used for Meir (or possibly other purposes). See #2349 that talks about this angle for Meier. Since it could be used for either one (or both) there needs to be one way to turn it on for either. It's looking like we won't be implementing this though for Meier at least in the near future. But, since this is a general dataset that could be used for other purposes it makes sense to not have it completely tied to just the dust emissions scheme.